impersonation/README.md

```diff
@@ -100,4 +101,104 @@ Creates a new impersonation request.
+
+- **URL:** `https://api.realdevsquad.com/impersonation/requests`
+
+- **Method:** GET
+
+- **Query Parameters:**
+
+  - `dev`: Required boolean to fetch requests.
+  - `id`: Optional string to fetch the request by its id.
+  - `page`: Optional integer to specify the page number.
+  - `size`: Optional integer to specify the number of requests per page. Default is 5.
+  - `createdBy`: Optional string to filter requests by username of super-user who created the request.
+  - `createdFor`: Optional string to filter requests by username of user for whom the request is created.
+  - `status`: Optional string to filter requests by status (e.g., APPROVED, REJECTED, PENDING).
+  - `prev`: Optional string to filter requests by the prev cursor, used in cursor based pagination.
+  - `next`: Optional string to filter requests by the next cursor, used in cursor based pagination.
+
+- **Headers:**
+
+  - Content-Type: application/json
+
+- **Cookie:**
+
+  - rds-session: `<JWT>`
+
+- **Success Response:**
+
+  - **Code:** 200
+  - **Content:**
+  
+    ```json
+    {
+      "message": "Request fetched successfully",
+      "data": [
+        {
+          "id": "string",
+          "createdAt": "Timestamp",
+          "updatedAt": "Timestamp",
+          "status": "string",
+          "userId": "string",
+          "impersonatedUserId": "string",
+          "isImpersonationFinished": "boolean",
+          "createdBy": "string",
+          "createdFor": "string",
+          "startedAt": "Timestamp",
+          "endedAt": "Timestamp",
+          "reason": "string"
+        },
+        {
+          "id": "string",
+          "createdAt": "Timestamp",
+          "updatedAt": "Timestamp",
+          "status": "string",
+          "userId": "string",
+          "impersonatedUserId": "string",
+          "isImpersonationFinished": "boolean",
+          "createdBy": "string",
+          "createdFor": "string",
+          "startedAt": "Timestamp",
+          "endedAt": "Timestamp",
+          "reason": "string"
+        }
+      ],
+      "next": "string",
+      "prev": "string",
+      "count": "number"
+    }
+    ```
+
+- **Error Responses:**
+
+  - **Code:** 500
+    - **Content:** `{ "statusCode": 500, "error": "Internal Server Error", "message": "An internal server error occurred" }`
+
+- **No Content Response:**
+
+  - **Code:** 204
+    - **Content:** No content
+
+
+#### Authentication and Authorization
+
+- Authentication is required for accessing this endpoint.
+
+#### Additional Notes
+
+- The provided response includes details of each request, such as ID, timestamps, username, userId, reason, status, etc.
+- Pagination functionality is implemented using `next` and `prev` parameters in the response.
+- Filtering options are available using parameters like `createdBy`, `createdFor`, `status`, etc.
+- The response includes a list of request objects with their respective properties.
+- Error handling is provided for internal server errors (status code 500).
```

Comment:
Here would want Specific details, Since this is a API contract
Please avoid ambiguity here
Specify only the allowed filters, be specific with the response etc please